app/gui2/src/composables/selection.ts

```diff
@@ -3,19 +3,21 @@ import { selectionMouseBindings } from '@/bindings'
 import { useEvent, usePointer } from '@/composables/events'
 import type { PortId } from '@/providers/portInfo.ts'
 import { type NodeId } from '@/stores/graph'
+import { filter, filterDefined, map } from '@/util/data/iterable'
 import type { Rect } from '@/util/data/rect'
 import { intersectionSize } from '@/util/data/set'
 import type { Vec2 } from '@/util/data/vec2'
 import { dataAttribute, elementHierarchy } from '@/util/dom'
 import * as set from 'lib0/set'
-import { filter } from 'shared/util/data/iterable'
 import { computed, ref, shallowReactive, shallowRef } from 'vue'
 
-export function useSelection<T>(
+export function useSelection<T, PackedT>(
   navigator: { sceneMousePos: Vec2 | null; scale: number },
   elementRects: Map<T, Rect>,
   margin: number,
   isValid: (element: T) => boolean,
+  pack: (element: T) => PackedT | undefined,
+  unpack: (packed: PackedT) => T | undefined,
```

Comment:
1. Add docs explaining what exactly is packed/unpacked value (with real-life example).
2. This starts having too many parameters. I think all optionals (pack, unpack and callbacks) should be in `Options` structure. margin also could be there, defaulting to 0